crypto/dilithium/pqcrystals_dilithium_ref_common/params.h

```diff
@@ -11,70 +15,42 @@
 #define Q 8380417
 #define D 13
 #define ROOT_OF_UNITY 1753
-
-#if DILITHIUM_MODE == 2
-#define K 4
-#define L 4
-#define ETA 2
-#define TAU 39
-#define BETA 78
-#define GAMMA1 (1 << 17)
-#define GAMMA2 ((Q-1)/88)
-#define OMEGA 80
-#define CTILDEBYTES 32
-
-#elif DILITHIUM_MODE == 3
-#define K 6
-#define L 5
-#define ETA 4
-#define TAU 49
-#define BETA 196
-#define GAMMA1 (1 << 19)
-#define GAMMA2 ((Q-1)/32)
-#define OMEGA 55
-#define CTILDEBYTES 48
-
-#elif DILITHIUM_MODE == 5
-#define K 8
-#define L 7
-#define ETA 2
-#define TAU 60
-#define BETA 120
-#define GAMMA1 (1 << 19)
-#define GAMMA2 ((Q-1)/32)
-#define OMEGA 75
-#define CTILDEBYTES 64
-
-#endif
-
 #define POLYT1_PACKEDBYTES  320
 #define POLYT0_PACKEDBYTES  416
-#define POLYVECH_PACKEDBYTES (OMEGA + K)
-
-#if GAMMA1 == (1 << 17)
-#define POLYZ_PACKEDBYTES   576
-#elif GAMMA1 == (1 << 19)
-#define POLYZ_PACKEDBYTES   640
-#endif
-
-#if GAMMA2 == (Q-1)/88
-#define POLYW1_PACKEDBYTES  192
-#elif GAMMA2 == (Q-1)/32
-#define POLYW1_PACKEDBYTES  128
-#endif
-
-#if ETA == 2
-#define POLYETA_PACKEDBYTES  96
-#elif ETA == 4
-#define POLYETA_PACKEDBYTES 128
-#endif
 
-#define CRYPTO_PUBLICKEYBYTES (SEEDBYTES + K*POLYT1_PACKEDBYTES)
-#define CRYPTO_SECRETKEYBYTES (2*SEEDBYTES \
-+ TRBYTES \
-+ L*POLYETA_PACKEDBYTES \
-+ K*POLYETA_PACKEDBYTES \
-+ K*POLYT0_PACKEDBYTES)
-#define CRYPTO_BYTES (CTILDEBYTES + L*POLYZ_PACKEDBYTES + POLYVECH_PACKEDBYTES)
+// Structure for ML-DSA parameters that depend on the parameter set.
+typedef struct {
```

Comment:
If this was all defined statically it could also all be const and avoid any issues where something accidentally changes one of these values.

Reply:
Andrew and I met in person to discuss the details of this PR, I will note the discussions here:

The reason for creating these structures are to form some code consistency with how we deal with the same issue in ML-KEM. See https://github.com/aws/aws-lc/blob/main/crypto/fipsmodule/ml_kem/ml_kem_ref/params.h#L23 for the respective part of the ML-KEM code. While this may not be the most efficient way to parse/point, I'd rather keep the code readable and somewhat analogous between the two algorithms. If we decide upon a design change for ML-DSA parameter selection, I'd like to see the same change in ML-KEM.

Reply:
if you make it const and define statically you'll have to deal with `delocate` when moving to the FIPS module, which is the main reason I didn't make it const in ML-KEM.